intermittent CI failures due to test discovery timeout

Occasionally, windows builds fail with timeout during test discovery, see e.g. https://github.com/DLR-SC/memilio/runs/4919449210?check_suite_focus=true

error message:
```
cmd.exe /C "cd . && "C:\Program Files\CMake\bin\cmake.exe" -E vs_link_exe --intdir=tests\CMakeFiles\memilio-test.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100220~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100220~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\ENTERP~1\VC\Tools\MSVC\1429~1.301\bin\Hostx64\x64\link.exe /nologo tests\CMakeFiles\memilio-test.dir\test_household.cpp.obj tests\CMakeFiles\memilio-test.dir\testmain.cpp.obj tests\CMakeFiles\memilio-test.dir\test_populations.cpp.obj tests\CMakeFiles\memilio-test.dir\test_seir_js_compare.cpp.obj tests\CMakeFiles\memilio-test.dir\test_numericalIntegration.cpp.obj tests\CMakeFiles\memilio-test.dir\test_smoother.cpp.obj tests\CMakeFiles\memilio-test.dir\test_damping.cpp.obj tests\CMakeFiles\memilio-test.dir\test_secir.cpp.obj tests\CMakeFiles\memilio-test.dir\test_implicit_euler.cpp.obj tests\CMakeFiles\memilio-test.dir\test_mobility.cpp.obj tests\CMakeFiles\memilio-test.dir\test_date.cpp.obj tests\CMakeFiles\memilio-test.dir\test_eigen_util.cpp.obj tests\CMakeFiles\memilio-test.dir\test_secir_ageres.cpp.obj tests\CMakeFiles\memilio-test.dir\test_parameter_studies.cpp.obj tests\CMakeFiles\memilio-test.dir\test_graph.cpp.obj tests\CMakeFiles\memilio-test.dir\test_graph_simulation.cpp.obj tests\CMakeFiles\memilio-test.dir\test_stl_util.cpp.obj tests\CMakeFiles\memilio-test.dir\test_uncertain.cpp.obj tests\CMakeFiles\memilio-test.dir\test_time_series.cpp.obj tests\CMakeFiles\memilio-test.dir\test_abm.cpp.obj tests\CMakeFiles\memilio-test.dir\test_analyze_result.cpp.obj tests\CMakeFiles\memilio-test.dir\test_contact_matrix.cpp.obj tests\CMakeFiles\memilio-test.dir\test_type_safe.cpp.obj tests\CMakeFiles\memilio-test.dir\test_custom_index_array.cpp.obj tests\CMakeFiles\memilio-test.dir\test_parameter_set.cpp.obj tests\CMakeFiles\memilio-test.dir\test_matrix_shape.cpp.obj tests\CMakeFiles\memilio-test.dir\test_damping_sampling.cpp.obj tests\CMakeFiles\memilio-test.dir\test_dynamic_npis.cpp.obj tests\CMakeFiles\memilio-test.dir\test_regions.cpp.obj tests\CMakeFiles\memilio-test.dir\test_io_framework.cpp.obj tests\CMakeFiles\memilio-test.dir\test_compartmentsimulation.cpp.obj tests\CMakeFiles\memilio-test.dir\test_mobility_io.cpp.obj tests\CMakeFiles\memilio-test.dir\test_transform_iterator.cpp.obj tests\CMakeFiles\memilio-test.dir\distributions_helpers.cpp.obj tests\CMakeFiles\memilio-test.dir\test_json_serializer.cpp.obj tests\CMakeFiles\memilio-test.dir\test_save_parameters.cpp.obj tests\CMakeFiles\memilio-test.dir\test_save_results.cpp.obj  /out:tests\memilio-test.exe /implib:tests\memilio-test.lib /pdb:tests\memilio-test.pdb /version:0.0 /machine:x64 /debug /INCREMENTAL /subsystem:console  memilio\memilio.lib  models\secir\secir.lib  models\seir\seir.lib  models\abm\abm.lib  lib\gtest_maind.lib  lib\gtestd.lib  lib\gmockd.lib  lib\gtest_maind.lib  memilio\memilio.lib  _deps\spdlog-build\spdlogd.lib  boost_filesystem.lib  "C:\Program Files\HDF_Group\HDF5\1.12.1\lib\libhdf5.lib"  "C:\Program Files\HDF_Group\HDF5\1.12.1\lib\libzlib.lib"  "C:\Program Files\HDF_Group\HDF5\1.12.1\lib\libsz.lib"  "C:\Program Files\HDF_Group\HDF5\1.12.1\lib\libaec.lib"  _deps\jsoncpp-build\src\lib_json\jsoncpp.lib  lib\gtestd.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cmd.exe /C "cd /D D:\a\memilio\memilio\cpp\build-win\tests && "C:\Program Files\CMake\bin\cmake.exe" -D TEST_TARGET=memilio-test -D TEST_EXECUTABLE=D:/a/memilio/memilio/cpp/build-win/tests/memilio-test.exe -D TEST_EXECUTOR= -D TEST_WORKING_DIR=D:/a/memilio/memilio/cpp/build-win/tests -D TEST_EXTRA_ARGS= -D TEST_PROPERTIES=FOLDER;Tests -D TEST_PREFIX=memilio-test. -D TEST_SUFFIX= -D TEST_FILTER= -D NO_PRETTY_TYPES=FALSE -D NO_PRETTY_VALUES=FALSE -D TEST_LIST=memilio-test_TESTS -D CTEST_FILE=D:/a/memilio/memilio/cpp/build-win/tests/memilio-test[1]_tests.cmake -D TEST_DISCOVERY_TIMEOUT=5 -D TEST_XML_OUTPUT_DIR= -P "C:/Program Files/CMake/share/cmake-3.22/Modules/GoogleTestAddTests.cmake"""
CMake Error at C:/Program Files/CMake/share/cmake-3.22/Modules/GoogleTestAddTests.cmake:83 (message):
  Error running test executable.

    Path: 'D:/a/memilio/memilio/cpp/build-win/tests/memilio-test.exe'
    Result: Process terminated due to timeout
    Output:
      

Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.22/Modules/GoogleTestAddTests.cmake:179 (gtest_discover_tests_impl)
```

--- a/cpp/tests/cmake/AddGoogleTest.cmake
+++ b/cpp/tests/cmake/AddGoogleTest.cmake
@@ -73,7 +73,8 @@ macro(add_gtest TESTNAME)
         else()
             gtest_discover_tests(${TESTNAME}
                 TEST_PREFIX "${TESTNAME}."
-                PROPERTIES FOLDER "Tests")
+                PROPERTIES FOLDER "Tests"
+                DISCOVERY_TIMEOUT 30)
         endif()
     else()
         add_test(${TESTNAME} ${TESTNAME})

